compiler/desugared/dependency.ml

```diff
@@ -58,19 +65,22 @@ module Vertex = struct
     | Var (x, sx), Var (y, sy) ->
       ScopeVar.equal x y && Option.equal StateName.equal sx sy
     | SubScope x, SubScope y -> SubScopeName.equal x y
-    | (Var _ | SubScope _), _ -> false
+    | Assertion a, Assertion b -> Ast.AssertionName.equal a b
+    | _, _ -> false
```

Comment:
```suggestion
    | (Var _ | SubScope _ | Assertion _), _ -> false
```
I prefer to be conservative for these, although it's a bit redudant, because otherwise you won't get warned when a new case is added.